### PR TITLE
API-diff between 7.0-preview4 and 7.0-preview5

### PR DESCRIPTION
**Pull request by carlossanlop**
The AsmDiff tool is unable to detect new attributes decorating types. I'm trying to find the bug and fix it. If you know of any new attribute that is not showing up, please add it as a suggestion in a comment.

ASP.NET: https://github.com/orgs/dotnet/teams/aspnet-api-review

WinForms: @merriemcgaw @RussKie

WPF: @dreddy-work @RussKie

Libraries: @danmoseley @jeffhandley @karelz @ericstj @stephentoub

Microsoft.VisualBasic @jaredpar @cston @333fred
System.Collections @dotnet/area-system-collections 
System.ComponentModel @dotnet/area-system-componentmodel 
System.Data @ajcvickers @DavoudEshtehari @David-Engel 
System.Diagnostics @dotnet/area-system-diagnostics-activity and @dotnet/area-system-diagnostics-process 
System.IO @dotnet/area-system-io
System.Net @dotnet/ncl 
System.Numerics @dotnet/area-system-numerics 
System.Reflection @dotnet/area-system-reflection 
System.Reflection.Metadata @dotnet/area-system-reflection-metadata 
System.Runtime.CompilerServices @dotnet/area-system-runtime-compilerservices 
Syste.Runtime.InteropServices @dotnet/interop-contrib 
System.Runtime.Intrinsics @dotnet/area-system-runtime-intrinsics 
System.Security @dotnet/area-system-security 
System.Text.Json @dotnet/area-system-text-json 
System.Threading @kouvel 
System @dotnet/area-system-runtime

### CODE REVIEW

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.ComponentModel.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# System.ComponentModel
+
+``` diff
+ namespace System.ComponentModel {
+     public sealed class TypeDescriptor {
+         public static Type ComObjectType { get; }
+         public static Type InterfaceType { get; }
```
Comment (eerhardt) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.ComponentModel.md:7]: These APIs just got "[EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Advanced)]" attributes put on them. They are not new.

https://github.com/dotnet/runtime/pull/68432/files#diff-6466df6aea2f15578eae754246e298f9fd3739cc4e81eb7736977ba7efde18e0R1387

Is there a bug in the diff tool here that it should be showing the new attributes, and not indicating these APIs are completely new?

cc @ericstj who made the PR

Comment (carlossanlop) [reply release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.ComponentModel.md:7]: Yep, I mentioned it in the description. 🙂

Comment (carlossanlop) [reply release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.ComponentModel.md:7]: I'll have to manually edit these.

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.DataProtection.AuthenticatedEncryption.ConfigurationModel.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Microsoft.AspNetCore.DataProtection.AuthenticatedEncryption.ConfigurationModel
+
+``` diff
+ namespace Microsoft.AspNetCore.DataProtection.AuthenticatedEncryption.ConfigurationModel {
+     public sealed class AuthenticatedEncryptorDescriptorDeserializer : IAuthenticatedEncryptorDescriptorDeserializer {
```
Comment (BrennanConroy) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.DataProtection.AuthenticatedEncryption.ConfigurationModel.md:5]:
```suggestion
-    [RequiresUnreferencedCode(TrimmerWarning.Message)]
     public sealed class AuthenticatedEncryptorDescriptorDeserializer : IAuthenticatedEncryptorDescriptorDeserializer {
```

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.DataProtection.AuthenticatedEncryption.ConfigurationModel.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+         public IAuthenticatedEncryptorDescriptor ImportFromXml(XElement element);
+     }
+     public sealed class CngCbcAuthenticatedEncryptorDescriptorDeserializer : IAuthenticatedEncryptorDescriptorDeserializer {
+         public IAuthenticatedEncryptorDescriptor ImportFromXml(XElement element);
```
Comment (BrennanConroy) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.DataProtection.AuthenticatedEncryption.ConfigurationModel.md:9]:
```suggestion
-        [RequiresUnreferencedCode(TrimmerWarning.Message)]
         public IAuthenticatedEncryptorDescriptor ImportFromXml(XElement element);
```

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.DataProtection.AuthenticatedEncryption.ConfigurationModel.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+     }
+     public sealed class CngGcmAuthenticatedEncryptorDescriptorDeserializer : IAuthenticatedEncryptorDescriptorDeserializer {
+         public IAuthenticatedEncryptorDescriptor ImportFromXml(XElement element);
```
Comment (BrennanConroy) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.DataProtection.AuthenticatedEncryption.ConfigurationModel.md:12]:
```suggestion
-        [RequiresUnreferencedCode(TrimmerWarning.Message)]
         public IAuthenticatedEncryptorDescriptor ImportFromXml(XElement element);
```

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.DataProtection.AuthenticatedEncryption.ConfigurationModel.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+     }
+     public interface IAuthenticatedEncryptorDescriptorDeserializer {
+         IAuthenticatedEncryptorDescriptor ImportFromXml(XElement element);
```
Comment (BrennanConroy) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.DataProtection.AuthenticatedEncryption.ConfigurationModel.md:15]:
```suggestion
-        [RequiresUnreferencedCode(TrimmerWarning.Message)]
         IAuthenticatedEncryptorDescriptor ImportFromXml(XElement element);
```

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.DataProtection.AuthenticatedEncryption.ConfigurationModel.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+     }
+     public sealed class ManagedAuthenticatedEncryptorDescriptorDeserializer : IAuthenticatedEncryptorDescriptorDeserializer {
+         public IAuthenticatedEncryptorDescriptor ImportFromXml(XElement element);
```
Comment (BrennanConroy) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.DataProtection.AuthenticatedEncryption.ConfigurationModel.md:18]:
```suggestion
-        [RequiresUnreferencedCode(TrimmerWarning.Message)]
         public IAuthenticatedEncryptorDescriptor ImportFromXml(XElement element);
```

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.DataProtection.Internal.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
+# Microsoft.AspNetCore.DataProtection.Internal
+
+``` diff
+ namespace Microsoft.AspNetCore.DataProtection.Internal {
+     public interface IActivator {
+         object CreateInstance(Type expectedBaseType, string implementationTypeName);
```
Comment (BrennanConroy) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.DataProtection.Internal.md:6]:
```suggestion
-        [RequiresUnreferencedCode(TrimmerWarning.Message)]
-        object CreateInstance(Type expectedBaseType, string implementationTypeName);
+        object CreateInstance([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] Type expectedBaseType, string implementationTypeName);
```

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.DataProtection.KeyManagement.Internal.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
+# Microsoft.AspNetCore.DataProtection.KeyManagement.Internal
+
+``` diff
+ namespace Microsoft.AspNetCore.DataProtection.KeyManagement.Internal {
+     public interface ICacheableKeyRingProvider {
+         CacheableKeyRing GetCacheableKeyRing(DateTimeOffset now);
```
Comment (BrennanConroy) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.DataProtection.KeyManagement.Internal.md:6]:
```suggestion
-        [RequiresUnreferencedCode(TrimmerWarning.Message)]
         CacheableKeyRing GetCacheableKeyRing(DateTimeOffset now);
```

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.DataProtection.KeyManagement.Internal.md [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+     }
+     public interface IInternalXmlKeyManager {
+         IAuthenticatedEncryptorDescriptor DeserializeDescriptorFromKeyElement(XElement keyElement);
```
Comment (BrennanConroy) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.DataProtection.KeyManagement.Internal.md:9]:
```suggestion
-        [RequiresUnreferencedCode(TrimmerWarning.Message)]
         IAuthenticatedEncryptorDescriptor DeserializeDescriptorFromKeyElement(XElement keyElement);
```

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.DataProtection.KeyManagement.Internal.md [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+     }
+     public interface IKeyRingProvider {
+         IKeyRing GetCurrentKeyRing();
```
Comment (BrennanConroy) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.DataProtection.KeyManagement.Internal.md:12]:
```suggestion
-        [RequiresUnreferencedCode(TrimmerWarning.Message)]
         IKeyRing GetCurrentKeyRing();
```

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.DataProtection.KeyManagement.md [hunk 11]
```diff
@@ -0,0 +1,14 @@
+# Microsoft.AspNetCore.DataProtection.KeyManagement
+
+``` diff
+ namespace Microsoft.AspNetCore.DataProtection.KeyManagement {
+     public interface IKeyManager {
+         IReadOnlyCollection<IKey> GetAllKeys();
```
Comment (BrennanConroy) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.DataProtection.KeyManagement.md:6]:
```suggestion
-        [RequiresUnreferencedCode(TrimmerWarning.Message)]
         IReadOnlyCollection<IKey> GetAllKeys();
```

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.DataProtection.KeyManagement.md [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+     }
+     public sealed class XmlKeyManager : IInternalXmlKeyManager, IKeyManager {
+         public IReadOnlyCollection<IKey> GetAllKeys();
```
Comment (BrennanConroy) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.DataProtection.KeyManagement.md:9]:
```suggestion
-        [RequiresUnreferencedCode(TrimmerWarning.Message)]
         public IReadOnlyCollection<IKey> GetAllKeys();
```

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.DataProtection.KeyManagement.md [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+         IAuthenticatedEncryptorDescriptor IInternalXmlKeyManager.DeserializeDescriptorFromKeyElement(XElement keyElement);
```
Comment (BrennanConroy) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.DataProtection.KeyManagement.md:10]:
```suggestion
-        [RequiresUnreferencedCode(TrimmerWarning.Message)]
         IAuthenticatedEncryptorDescriptor IInternalXmlKeyManager.DeserializeDescriptorFromKeyElement(XElement keyElement);
```

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.DataProtection.md [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+     public interface IDataProtector : IDataProtectionProvider {
+         byte[] Protect(byte[] plaintext);
+         byte[] Unprotect(byte[] protectedData);
+     }
+     public interface IPersistedDataProtector : IDataProtectionProvider, IDataProtector {
+         byte[] DangerousUnprotect(byte[] protectedData, bool ignoreRevocationErrors, out bool requiresMigration, out bool wasRevoked);
+     }
+     public interface ITimeLimitedDataProtector : IDataProtectionProvider, IDataProtector {
+         byte[] Protect(byte[] plaintext, DateTimeOffset expiration);
+         byte[] Unprotect(byte[] protectedData, out DateTimeOffset expiration);
```
Comment (BrennanConroy) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.DataProtection.md:24]:
```suggestion
-        [RequiresUnreferencedCode(TrimmerWarning.Message)]
         public static byte[] Protect(this ITimeLimitedDataProtector protector, byte[] plaintext, TimeSpan lifetime);
-        [RequiresUnreferencedCode(TrimmerWarning.Message)]
         public static string Protect(this ITimeLimitedDataProtector protector, string plaintext, DateTimeOffset expiration);
-        [RequiresUnreferencedCode(TrimmerWarning.Message)]
         public static string Protect(this ITimeLimitedDataProtector protector, string plaintext, TimeSpan lifetime);
-        [RequiresUnreferencedCode(TrimmerWarning.Message)]
         public static string Unprotect(this ITimeLimitedDataProtector protector, string protectedData, out DateTimeOffset expiration);
     }
     public static class DataProtectionCommonExtensions {
-        [RequiresUnreferencedCode(TrimmerWarning.Message)]
         public static string Protect(this IDataProtector protector, string plaintext);
-        [RequiresUnreferencedCode(TrimmerWarning.Message)]
         public static string Unprotect(this IDataProtector protector, string protectedData);
     }
     public interface IDataProtector : IDataProtectionProvider {
-        [RequiresUnreferencedCode(TrimmerWarning.Message)]
         byte[] Protect(byte[] plaintext);
-        [RequiresUnreferencedCode(TrimmerWarning.Message)]
         byte[] Unprotect(byte[] protectedData);
     }
     public interface IPersistedDataProtector : IDataProtectionProvider, IDataProtector {
-        [RequiresUnreferencedCode(TrimmerWarning.Message)]
         byte[] DangerousUnprotect(byte[] protectedData, bool ignoreRevocationErrors, out bool requiresMigration, out bool wasRevoked);
     }
     public interface ITimeLimitedDataProtector : IDataProtectionProvider, IDataProtector {
-        [RequiresUnreferencedCode(TrimmerWarning.Message)]
         byte[] Protect(byte[] plaintext, DateTimeOffset expiration);
-        [RequiresUnreferencedCode(TrimmerWarning.Message)]
         byte[] Unprotect(byte[] protectedData, out DateTimeOffset expiration);
```

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.Http.HttpResults.md [hunk 15]
```diff
@@ -0,0 +1,16 @@
+# Microsoft.AspNetCore.Http.HttpResults
+
+``` diff
+ namespace Microsoft.AspNetCore.Http.HttpResults {
```
Comment (BrennanConroy) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.Http.HttpResults.md:4]:
```suggestion
- namespace Microsoft.AspNetCore.Http {
 namespace Microsoft.AspNetCore.Http.HttpResults {
```

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.ResponseCompression.md [hunk 16]
```diff
@@ -0,0 +1,10 @@
+# Microsoft.AspNetCore.ResponseCompression
+
+``` diff
+ namespace Microsoft.AspNetCore.ResponseCompression {
+     public class CompressionProviderCollection : Collection<ICompressionProvider> {
+         public void Add(Type providerType);
```
Comment (BrennanConroy) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.AspNetCore.ResponseCompression.md:6]:
```suggestion
-         public void Add(Type providerType);
+         public void Add([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] Type providerType);
```

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.JSInterop.Infrastructure.md [hunk 17]
```diff
@@ -0,0 +1,8 @@
+# Microsoft.JSInterop.Infrastructure
+
+``` diff
+ namespace Microsoft.JSInterop.Infrastructure {
+     public static class DotNetDispatcher
```
Comment (BrennanConroy) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.JSInterop.Infrastructure.md:5]: This one is odd, did the tool break?
I assume it's trying to show the attribute changes from https://github.com/dotnet/aspnetcore/commit/442a380854c25b85252a743c3ba2e21f08fa6879#diff-ee83750b0c4b366580d485516093fb99c3c9b1fce10eef52846211bcb9e9021e

Comment (carlossanlop) [reply release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.JSInterop.Infrastructure.md:5]: > I assume it's trying to show the attribute changes

Yes, that's what happened. It's missing the attribute change, and it also did not add the empty `{}` to indicate no members were modified.

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.Numerics.md [hunk 18]
```diff
@@ -0,0 +1,239 @@
...
++        public static bool IsPow2(BigInteger value);
++        public static BigInteger LeadingZeroCount(BigInteger value);
++        public static BigInteger Log2(BigInteger value);
++        public static BigInteger MaxMagnitude(BigInteger x, BigInteger y);
++        public static BigInteger MinMagnitude(BigInteger x, BigInteger y);
++        public static explicit operator BigInteger (Half value);
++        public static explicit operator char (BigInteger value);
++        public static explicit operator Half (BigInteger value);
++        public static explicit operator Int128 (BigInteger value);
++        public static explicit operator IntPtr (BigInteger value);
```
Comment (tannergooding) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.Numerics.md:27]:
```suggestion
+        public static explicit operator nint (BigInteger value);
```

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.Numerics.md [hunk 19]
```diff
@@ -0,0 +1,239 @@
...
++        public static explicit operator UInt128 (BigInteger value);
++        public static explicit operator UIntPtr (BigInteger value);
```
Comment (tannergooding) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.Numerics.md:29]:
```suggestion
+        public static explicit operator nuint (BigInteger value);
```

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.md [hunk 20]
```diff
@@ -0,0 +1,1118 @@
...
++        public static Int128 operator checked +(Int128 left, Int128 right);
++        public static Int128 operator checked --(Int128 value);
++        public static Int128 operator checked /(Int128 left, Int128 right);
++        public static explicit operator checked Int128 (double value);
++        public static explicit operator checked byte (Int128 value);
++        public static explicit operator checked char (Int128 value);
++        public static explicit operator checked short (Int128 value);
++        public static explicit operator checked int (Int128 value);
++        public static explicit operator checked long (Int128 value);
++        public static explicit operator checked IntPtr (Int128 value);
```
Comment (tannergooding) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.md:365]:
```suggestion
+        public static explicit operator checked nint (Int128 value);
```

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.md [hunk 21]
```diff
@@ -0,0 +1,1118 @@
...
++        public static explicit operator Int128 (double value);
++        public static explicit operator byte (Int128 value);
++        public static explicit operator char (Int128 value);
++        public static explicit operator decimal (Int128 value);
++        public static explicit operator double (Int128 value);
++        public static explicit operator Half (Int128 value);
++        public static explicit operator short (Int128 value);
++        public static explicit operator int (Int128 value);
++        public static explicit operator long (Int128 value);
++        public static explicit operator IntPtr (Int128 value);
```
Comment (tannergooding) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.md:391]: Many of these are duplicated in here.

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.md [hunk 22]
```diff
@@ -0,0 +1,1118 @@
...
++        public static explicit operator sbyte (Int128 value);
++        public static explicit operator float (Int128 value);
++        public static explicit operator UInt128 (Int128 value);
++        public static explicit operator ushort (Int128 value);
++        public static explicit operator uint (Int128 value);
++        public static explicit operator ulong (Int128 value);
++        public static explicit operator UIntPtr (Int128 value);
```
Comment (tannergooding) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.md:398]:
```suggestion
+        public static explicit operator nuint (Int128 value);
```

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.md [hunk 23]
```diff
@@ -0,0 +1,1118 @@
...
++        public static explicit operator Int128 (float value);
++        public static bool operator >(Int128 left, Int128 right);
++        public static bool operator >=(Int128 left, Int128 right);
++        public static implicit operator Int128 (byte value);
++        public static implicit operator Int128 (char value);
++        public static implicit operator Int128 (short value);
++        public static implicit operator Int128 (int value);
++        public static implicit operator Int128 (long value);
++        public static implicit operator Int128 (IntPtr value);
```
Comment (tannergooding) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.md:407]:
```suggestion
+        public static implicit operator Int128 (nint value);
```

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.md [hunk 24]
```diff
@@ -0,0 +1,1118 @@
...
++        public static implicit operator Int128 (sbyte value);
++        public static implicit operator Int128 (ushort value);
++        public static implicit operator Int128 (uint value);
++        public static implicit operator Int128 (ulong value);
++        public static implicit operator Int128 (UIntPtr value);
```
Comment (tannergooding) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.md:412]:
```suggestion
+        public static implicit operator Int128 (nuint value);
```

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.md [hunk 25]
```diff
@@ -0,0 +1,1118 @@
...
++        public static UInt128 operator &(UInt128 left, UInt128 right);
++        public static UInt128 operator |(UInt128 left, UInt128 right);
++        public static UInt128 operator checked +(UInt128 left, UInt128 right);
++        public static UInt128 operator checked --(UInt128 value);
++        public static UInt128 operator checked /(UInt128 left, UInt128 right);
++        public static explicit operator checked UInt128 (double value);
++        public static explicit operator checked UInt128 (short value);
++        public static explicit operator checked UInt128 (int value);
++        public static explicit operator checked UInt128 (long value);
++        public static explicit operator checked UInt128 (IntPtr value);
```
Comment (tannergooding) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.md:792]:
```suggestion
+        public static explicit operator checked UInt128 (nint value);
```

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.md [hunk 26]
```diff
@@ -0,0 +1,1118 @@
...
++        public static explicit operator checked short (UInt128 value);
++        public static explicit operator checked int (UInt128 value);
++        public static explicit operator checked long (UInt128 value);
++        public static explicit operator checked Int128 (UInt128 value);
++        public static explicit operator checked IntPtr (UInt128 value);
++        public static explicit operator checked sbyte (UInt128 value);
++        public static explicit operator checked ushort (UInt128 value);
++        public static explicit operator checked uint (UInt128 value);
++        public static explicit operator checked ulong (UInt128 value);
++        public static explicit operator checked UIntPtr (UInt128 value);
```
Comment (tannergooding) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.md:806]:
```suggestion
+        public static explicit operator checked nuint (UInt128 value);
```

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.md [hunk 27]
```diff
@@ -0,0 +1,1118 @@
...
++        public static explicit operator short (UInt128 value);
++        public static explicit operator int (UInt128 value);
++        public static explicit operator long (UInt128 value);
++        public static explicit operator IntPtr (UInt128 value);
++        public static explicit operator sbyte (UInt128 value);
++        public static explicit operator float (UInt128 value);
++        public static explicit operator ushort (UInt128 value);
++        public static explicit operator uint (UInt128 value);
++        public static explicit operator ulong (UInt128 value);
++        public static explicit operator UIntPtr (UInt128 value);
```
Comment (tannergooding) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.md:838]:
```suggestion
+        public static explicit operator nuint (UInt128 value);
```

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.md [hunk 28]
```diff
@@ -0,0 +1,1118 @@
...
++        public static bool operator >(UInt128 left, UInt128 right);
++        public static bool operator >=(UInt128 left, UInt128 right);
++        public static implicit operator UInt128 (byte value);
++        public static implicit operator UInt128 (char value);
++        public static implicit operator UInt128 (ushort value);
++        public static implicit operator UInt128 (uint value);
++        public static implicit operator UInt128 (ulong value);
++        public static implicit operator UInt128 (UIntPtr value);
```
Comment (tannergooding) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.md:846]:
```suggestion
+        public static implicit operator UInt128 (nuint value);
```

**Review by tannergooding — APPROVED** [review]

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.Diagnostics.md [hunk 29]
```diff
@@ -0,0 +1,31 @@
...
+-        public Activity? StartActivity(string name, ActivityKind kind, ActivityContext parentContext, IEnumerable<KeyValuePair<string, object?>>? tags = null, IEnumerable<ActivityLink>? links = null, DateTimeOffset startTime = default(DateTimeOffset));
++        public Activity StartActivity(string name, ActivityKind kind, ActivityContext parentContext, IEnumerable<KeyValuePair<string, object?>>? tags = null, IEnumerable<ActivityLink> links = null, DateTimeOffset startTime = default(DateTimeOffset));
+-        public Activity? StartActivity(string name, ActivityKind kind, string parentId, IEnumerable<KeyValuePair<string, object?>>? tags = null, IEnumerable<ActivityLink>? links = null, DateTimeOffset startTime = default(DateTimeOffset));
++        public Activity StartActivity(string name, ActivityKind kind, string parentId, IEnumerable<KeyValuePair<string, object?>>? tags = null, IEnumerable<ActivityLink> links = null, DateTimeOffset startTime = default(DateTimeOffset));
+     }
```
Comment (tarekgh) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.Diagnostics.md:24]: This looks wrong changes here. This class didn't change at all.

[https://github.com/dotnet/runtime/blob/fededf1a376aa8e667ac8dd752d997308814df45/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/ActivitySource.cs#L55](https://github.com/dotnet/runtime/blob/fededf1a376aa8e667ac8dd752d997308814df45/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/ActivitySource.cs#L55)

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.JSInterop.Infrastructure.md (hunk 17 above)
Comment (carlossanlop) [reply release-notes/7.0/preview/api-diff/preview5/Microsoft.AspNetCore.App/7.0-preview5_Microsoft.JSInterop.Infrastructure.md:5]:
```suggestion
     public static class DotNetDispatcher
     {
+        [System.Diagnostics.CodeAnalysis.UnconditionalSuppressMessageAttribute("Trimming", "IL2026", Justification = "We expect application code is configured to ensure return types of JSInvokable methods are retained.")]
         public static string? Invoke(Microsoft.JSInterop.JSRuntime jsRuntime, in Microsoft.JSInterop.Infrastructure.DotNetInvocationInfo invocationInfo, string argsJson);
+        [System.Diagnostics.CodeAnalysis.UnconditionalSuppressMessageAttribute("Trimming", "IL2026", Justification = "We expect application code is configured to ensure return types of JSInvokable methods are retained.")]
         public static void BeginInvokeDotNet(Microsoft.JSInterop.JSRuntime jsRuntime, Microsoft.JSInterop.Infrastructure.DotNetInvocationInfo invocationInfo, string argsJson)
     }
```

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.ComponentModel.md (hunk 1 above)
Comment (carlossanlop) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.ComponentModel.md:7]:
```suggestion
+       [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Advanced)]
         public static Type ComObjectType { get; }
+       [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Advanced)]
         public static Type InterfaceType { get; }
```

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.Diagnostics.md (hunk 29 above)
Comment (carlossanlop) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.Diagnostics.md:24]:
```suggestion
```

Comment (carlossanlop) [reply release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.Diagnostics.md:24]: Removed the whole type. Thanks for checking.

**Comment by tarekgh** [comment]
`System.Diagnostics.ActivityEvent` and `System.Diagnostics.ActivityLink` looks good to me.

File: release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.Net.Security.md [hunk 30]
```diff
@@ -0,0 +1,13 @@
+# System.Net.Security
+
+``` diff
+ namespace System.Net.Security {
+     public readonly struct SslClientHelloInfo {
+         public string ServerName { get; }
+         public SslProtocols SslProtocols { get; }
+     }
+-    public sealed class SslStreamCertificateContext
++    public class SslStreamCertificateContext
```
Comment (stephentoub) [inline release-notes/7.0/preview/api-diff/preview5/Microsoft.NETCore.App/7.0-preview5_System.Net.Security.md:10]: @wfurt, SslStreamCertificateContext was unsealed?  Why?

**Comment by cston** [comment]
`Microsoft.VisualBasic` LGTM.

**Pull request merged**